### DISCUSSION TITLE
networking/ingress-nginx/high-availability

### DISCUSSION THREAD

**Discussion by giscus** [imroc/tke-guide#9] [General]
# networking/ingress-nginx/high-availability

概述

https://imroc.cc/tke/networking/ingress-nginx/high-availability

**Comment by ermazi** [comment]
配置自动扩缩容：那边配置注释写反了!

**Reply by ermazi** [reply]
打散调度 那边, 因为配置了 whenUnsatisfiable: ScheduleAnyway, 只是优选多打了几分, 权重不高的, 如果在混合负载的场景下, 并不能做到尽可能打散呢

**Reply by imroc** [reply]
> 配置自动扩缩容：那边配置注释写反了!

多谢反馈, 已改正

**Reply by imroc** [reply]
> 打散调度 那边, 因为配置了 whenUnsatisfiable: ScheduleAnyway, 只是优选多打了几分, 权重不高的, 如果在混合负载的场景下, 并不能做到尽可能打散呢

看做到什么程度, 文中给了例子, 可以微调。如果不考虑资源的情况, 自己改成 DoNotSchedule 就行

**Reply by ermazi** [reply]
"尽可能调度"单单靠这个配置是做不到的, 怎么调整都做不到~
"DoNotSchedule" 的问题在于单可用区大面积资源不可用的情况下如何保证容量的, 不敢直接用, 所以才需要满足"尽可能调度"

**Comment by ermazi** [comment]
文中提到专机专用, 超级节点是个解法, 但是引入超级节点本身约束还是蛮多的;假如要专机专用, 该如何配置保证呢(一节点一台, 资源尽可能给 ng, 配和停机节点池想用?)

**Reply by imroc** [reply]
假如要专机专用, 该如何配置保证呢(一节点一台, 资源尽可能给 ng, 配和停机节点池想用?)

如果不用超级节点, 也不考虑资源利用率, 可给节点池打污点, 只允许ng调度上去, ng 再用 拓扑分布约束强制单机单pod打散
